### ISSUE TITLE
GPL-838 new method of DNA quantification: Tapestation (C=M, V=?)

### ISSUE THREAD

**Issue by rl15** [sanger/quanthub#102]
**User story**
The clear and concise user story describing what is required. e.g. As an SSR user I would like sample manifests to allow me to upload x ...

**Who are the primary contacts for this story**
e.g. John S (don't include surnames in public repos)

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] add a list of acceptance criteria here
- [ ] ...

**Dependencies**
This story is blocked by the following dependencies:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

**Additional context**
Add any other context or screenshots about the feature request here.Stef wrote (Thursday, 14 November 2019 at 12:18)

Hi Emma and Rich,
 
We have a new method of DNA quantification: Tapestation.
We will be using this machine to quantify 10X VDJ libraries (as opposed to using qPCR or Mosquito).
We wish to be able to upload these quants into quanthub. To do this we would need to get quanthub to accept such files.
 
I have written this up as a research story. It is not urgent right now but we would like to be able to do this in the next few months.
It would be interesting to find out how much work this would require and how quickly it could be implemented.
 
Thank you and best wishes,

**Comment by stevieing** [comment]
Please find an example output file attached.
For me it would just be a case of outputting the region molarity value. This would then ideally be downloadable against the well ID and the pool (as it currently the case when you download ng/ul at the end of pipelines in Iimber.
Occasionally the DNA is diluted before it is run so a correction factor may be required e.g. *10.

**Comment by stevieing** [comment]
I (Stef) wish to be able to upload this csv file into quanthub. The region molarity (nmol/L) value should be exported to sequencescape. This information should then be available for download in limber (along with the well ID and pool).
The plate barcode can be provided both in the title of the file and in col A (as displayed). If desired it can also be a value in column C.
The well ID (as described in column B) corresponds to the well ID in the source plate.
The nmol/L value imported in the csv file is a 1/10 dilution, so upon export to sequencescpae this value needs to be multiplied by 10.
 
Acceptance criteria: csv file that comes off tapestation can be uploaded to quanthub. Quanthub exports nmol/L value to sequencescape and multiplies this value by 10. nM concentration can be downloaded for the plate in Limber alongside the well ID and pool.
[DNXXXXXX - 2019-11-05 - 16-32-56-D5000_compactRegionTable(1).xlsx](https://github.com/sanger/quanthub/files/6048542/DNXXXXXX.-.2019-11-05.-.16-32-56-D5000_compactRegionTable.1.xlsx)

**Comment by sdjmchattie** [comment]
There is an implementation in PR https://github.com/sanger/quanthub/pull/116 which achieves the criteria above, but during review further questions are being raised which were assumptions made during development that may or may not be correct:

* What should the assay be named as it is imported from Tapestation?
* Are there replicates in the Tapestation quantification?  Other systems are using a 384 well plate with 3 replicates from 96 well plates and an empty well.
  * If so, what metric should be used for the outlier detection?
  * How do we identify wells containing replicates?
* Is it right that we should be taking the region molarity from the Tapestation output in nmol/L and multiplying by 10 to account for dilution, then recording that as the concentration in nM?
* Could we get a full file output from the machine so we can ensure we've handled the complete data set correctly?
* Will the filenames be as shown above (DNXXXXXX - 2019-11-05 - 16-32-56-D5000_compactRegionTable(1).xlsx)?  Is this going to remain consistent?  As we're pulling the plate barcode from this filename.

I will attempt to organise a meeting with Stef to confirm.  If Stef is unsure, we'll wait for @stevieing to be back.

**Comment by sdjmchattie** [comment]
There is a branch with work towards this.  We thought it was needed sooner, but it wasn't needed yet. https://github.com/sanger/quanthub/tree/GPL-838-tapestation-support

**Comment by harrietc52** [comment]
Todo:
Contact Stef asking if this is still required

**Comment by stevieing** [comment]
Already done. I emailed Stef yesterday yesterday.

Quanthub already has tapestation support so we might be able to repurpose it.

On 9 Feb 2022, at 15:30, Harriet Craven ***@***.******@***.***>> wrote:



Todo:
Contact Stef asking if this is still required

—
Reply to this email directly, view it on GitHub [github.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sanger_quanthub_issues_102-23issuecomment-2D1033885531&d=DwMCaQ&c=D7ByGjS34AllFgecYw0iC6Zq7qlm8uclZFI0SqQnqBo&r=PioAYl_Zz9lnGwJRms8MsEB0E-grNRW-ut-_gAGl70g&m=dnnY2qRAJk4W-7ul7kS3ZuT-qIS7yftKn7Oh1rtXnr_suvpdkcl69xtZWHzX968c&s=7EpHZYuJ-ei88EC3nNGiKVMU-cdcBP3j-DkuUxrdH54&e=>, or unsubscribe [github.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AAD6ZH66P44BT4VWY5NVKITU2KCBZANCNFSM4YGHWN3Q&d=DwMCaQ&c=D7ByGjS34AllFgecYw0iC6Zq7qlm8uclZFI0SqQnqBo&r=PioAYl_Zz9lnGwJRms8MsEB0E-grNRW-ut-_gAGl70g&m=dnnY2qRAJk4W-7ul7kS3ZuT-qIS7yftKn7Oh1rtXnr_suvpdkcl69xtZWHzX968c&s=71Wbh-HDzZLG8_nnOSBdXUB9lAfc5GFjoKZwVM-tn2k&e=>.
Triage notifications on the go with GitHub Mobile for iOS [apps.apple.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__apps.apple.com_app_apple-2Dstore_id1477376905-3Fct-3Dnotification-2Demail-26mt-3D8-26pt-3D524675&d=DwMCaQ&c=D7ByGjS34AllFgecYw0iC6Zq7qlm8uclZFI0SqQnqBo&r=PioAYl_Zz9lnGwJRms8MsEB0E-grNRW-ut-_gAGl70g&m=dnnY2qRAJk4W-7ul7kS3ZuT-qIS7yftKn7Oh1rtXnr_suvpdkcl69xtZWHzX968c&s=ivwMlcOR5vdl7OvdBSb4IBqtFNuQ8dCOTGNNk4U4wOI&e=> or Android [play.google.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__play.google.com_store_apps_details-3Fid-3Dcom.github.android-26referrer-3Dutm-5Fcampaign-253Dnotification-2Demail-2526utm-5Fmedium-253Demail-2526utm-5Fsource-253Dgithub&d=DwMCaQ&c=D7ByGjS34AllFgecYw0iC6Zq7qlm8uclZFI0SqQnqBo&r=PioAYl_Zz9lnGwJRms8MsEB0E-grNRW-ut-_gAGl70g&m=dnnY2qRAJk4W-7ul7kS3ZuT-qIS7yftKn7Oh1rtXnr_suvpdkcl69xtZWHzX968c&s=zw5RE69qnWJ2ObEYR_M1nb1Rab7af99a4V0-53KbYBc&e=>.
You are receiving this because you were mentioned.Message ID: ***@***.***>




-- 
 The Wellcome Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.

**Comment by Skrich1999** [comment]
Rich C contacted Stef to evaluate the current position of this story.  

Stef replied:

Is this story a “nice to have”? (N)
Does this story contribute to the institutes OKRs? (N)
Is it essential this story is completed within the next 3 months (Business Critical)? (N)
If this story is not completed, what is this impact? (none)

We don’t need this right now. But rather than deleting it, it would be better to archive it. As it may become important in the future and so we would not lose all this information.

**Comment by Skrich1999** [comment]
Rich C - Closing to Archive this story
